runtime/parachains/src/disputes.rs

```diff
@@ -675,16 +671,10 @@ impl<BlockNumber: Clone> DisputeStateImporter<BlockNumber> {
 					.collect()
 			},
 			(false, true) => {
-				let prev_participants = {
-					// all participants
-					let mut a = self.state.validators_for.clone();
-					*a |= self.state.validators_against.iter().by_val();
-
-					// which are not new participants
-					*a &= self.new_participants.iter().by_val().map(|b| !b);
-
-					a
-				};
+				// all participants, which are not new participants
+				let prev_participants = (self.state.validators_for.clone() |
+					self.state.validators_against.clone()) &
+					!self.new_participants.clone();
```

Comment:
Now this was trickier, but I still think `self.new_participants` has the same number of bits that `self.state.validators_for` and `self.state.validators_against` has (see `let new_participants = bitvec::bitvec![u8, BitOrderLsb0; 0; state.validators_for.len()]` above).

There is also a chance that the whole algorithm gets simpler/faster if merged with the next `.iter_ones().map(...).collect()` transformation.

Reply:
I could replace `self.state.validators_against.clone()` with a borrow here, otherwise it matches the previous implementation.